src/main.F90

```diff
@@ -38,6 +38,11 @@ program main
 !
 !======================================================================
  call finalize()     ! Finalize computation
+
+ call deallocate_all()
+ call setup()
+ call solver()
+ call finalize()     ! Finalize computation
```

Comment:
proof of concept, not meant for real use. This shows that after stopping the simulation (the first `finalize`) you can 

* deallocate the arrays
* call `setup` which reallocates them
* run the solver
* stop again

and you will get the same data twice. Not sure if I ever proved it was the same data, but I do believe it ran without error.